Jenkinsfile

```diff
@@ -20,134 +20,197 @@ pipeline {
+              steps {
+                dir(path: "$REPO_DIR") {
+                  sh "make lint"
+                }
+              }
+            }
+            stage("make format") {
+              steps {
+                dir(path: "$REPO_DIR") {
+                  sh "make format"
+                }
+              }
+            }
+            stage("make vet") {
+              steps {
+                dir(path: "$REPO_DIR") {
+                  sh "make vet"
+                }
+              }
+            }
+            stage("make cyclomatic-check") {
+              steps {
+                dir(path: "$REPO_DIR") {
+                  sh "make cyclomatic-check"
+                }
+              }
+            }
+            stage("make test BUILDTAGS=kerneldrv") {
+              steps {
+                dir(path: "$REPO_DIR") {
+                  sh "make test BUILDTAGS=kerneldrv"
+                }
+              }
             }
           }
         }
-        stage("make vet") {
+        stage('make pre-pull') {
           steps {
-            dir(path: "$REPO_DIR") {
-              sh "make vet"
+    	      dir(path: "$REPO_DIR") {
+              sh "make pre-pull"
             }
           }
         }
-        stage("make cyclomatic-check") {
-          steps {
-            dir(path: "$REPO_DIR") {
-              sh "make cyclomatic-check"
+        stage('make images') {
+          parallel {
+            stage("make images with docker") {
+              steps {
+                dir(path: "$REPO_DIR") {
+                  sh "make -j4 images"
+                }
+              }
+            }
+            stage("make images with buildah") {
+              steps {
+                dir(path: "$REPO_DIR") {
+                  sh "make images BUILDER=buildah"
+                }
+              }
             }
           }
         }
-        stage("make test BUILDTAGS=kerneldrv") {
-          steps {
-            dir(path: "$REPO_DIR") {
-              sh "make test BUILDTAGS=kerneldrv"
+        stage('make demos') {
+          parallel {
+            stage('make demos with docker') {
+              steps {
+                dir(path: "$REPO_DIR") {
+                  sh "make demos"
+                }
+              }
+            }
+            stage('make demos with buildah') {
+              steps {
+                dir(path: "$REPO_DIR") {
+                  sh "make demos BUILDER=buildah"
+                }
+              }
             }
           }
         }
       }
-    }
-    stage('make pre-pull') {
-      steps {
-	dir(path: "$REPO_DIR") {
-          sh "make pre-pull"
+      post {
+        success {
+          withDockerRegistry([ credentialsId: "e16bd38a-76cb-4900-a5cb-7f6aa3aeb22d", url: "https://${REG}" ]) {
+            sh "make push"
+          }
         }
       }
     }
-    stage('make images') {
-      parallel {
-        stage("make images with docker") {
+    stage('Intel QAT plugin') {
```

Comment:
This flow and stages need thinking and discussion. 

Ideally, the flow is that Jenkinsfile asks for a cluster with all the necessary HW capabilities and runs the tests for each plugin that implements tests (this follows the pattern we have for building).  Something like `make test-cluster`

How to make the checks.sh and image pulling work if the test cluster has several worker nodes?

Reply:
In last commit I'm simplifying the stage for test qat as part of a several devices, to add fpga we only need to add a new stage in the parallel block (assuming by now this is a single node with all hw) and thats all, ... for a multi node cluster, as you mentioned is something that we need to think and think well since the complexity get a little bit higher.